packages/metal-soy/src/Soy.js

```diff
@@ -60,7 +62,12 @@ class Soy extends IncrementalDomRenderer.constructor {
 				data[params[i]] = component[params[i]].bind(component);
 			}
 		}
-		return data;
+
+		if (isFunction(component.prepareStateForRender)) {
```

Comment:
@mairatma Didn’t want to make `data` mutable, so I’m branching the return here. Tell me if this is ok with you.